src/components/basketPage/basketPage.ts

```diff
@@ -0,0 +1,328 @@
+import './basketPage.scss';
+import ModalPayment from '../modalPayment/modalPayment';
+import { IProducts, IBasketProduct, IFiltersElems, TPromo } from '../../models';
+import { createElem } from '../../utilities';
+
+export default class BasketPage {
+  productsItemNumElem: IFiltersElems = {};
+  promoCodes: TPromo = {
+    test: {
+      percent: 10,
+      isUsed: false,
+      description: 'Promo test',
+    },
+    rs: {
+      percent: 25,
+      isUsed: false,
+      description: 'Rolling Scopes School',
+    },
+  };
+
+  drawBasketPage(): void {
+    const main = document.querySelector('.main') as HTMLElement;
+    main.innerHTML = this.basketPageTemplate();
+
+    const basket = this.getBasketProductsArr();
+    this.productsItemNumElem = {};
+    const basketPageItems = document.querySelector('.basket-page__items') as HTMLElement;
+
+    basketPageItems.innerHTML = '';
+    if (!basket.length) {
+      createElem('div', 'empty-basket', basketPageItems, 'Card is empty');
+    } else {
+      basket.forEach((basketProduct, index) => {
+        const id = basketProduct.product.id;
+        const price = basketProduct.product.price;
+        const basketItemBlock = createElem('div', 'basket-page__item');
+        basketItemBlock.innerHTML = this.basketPageItemTemplate(basketProduct.product, index, basketProduct.amount);
+        const keyBlock = `${id}_block`;
+        this.productsItemNumElem[keyBlock] = basketItemBlock;
+
+        const productAmountElem = basketItemBlock.querySelector('.basket-product-info__amount') as HTMLElement;
+        const keyAmountElem = `${id}_amount`;
+        this.productsItemNumElem[keyAmountElem] = productAmountElem;
+
+        const removeBtn = basketItemBlock.querySelector('#remove-product-btn') as HTMLButtonElement;
+        const addBtn = basketItemBlock.querySelector('#add-product-btn') as HTMLButtonElement;
+
+        removeBtn.addEventListener('click', (e) => this.addOrRemoveProduct(e, id, price));
+        addBtn.addEventListener('click', (e) => this.addOrRemoveProduct(e, id, price));
+
+        const currAmountSum = basketItemBlock.querySelector('.basket-product-info__amount-sum') as HTMLElement;
+        const keyTotal = `${id}_total`;
+        this.productsItemNumElem[keyTotal] = currAmountSum;
+
+        basketPageItems.append(basketItemBlock);
+      });
+    }
+
+    this.summaryData();
+
+    // ==========
```

Comment:
PR should not include comments. 
If you want to leave a comment - that means your code isn't self-described enough, think about refactoring